### ISSUE TITLE
Improve accessibility of Facet widgets

### ISSUE THREAD

**Issue by JakePT** [10up/ElasticPress#2596]
**Is your enhancement related to a problem? Please describe.**
When discussing the accessibility of the Facet widget with the team a few issues were identified:

1. Filter links lack an accessible label the describes what will happen when the link is followed. For example, to assistive technologies the link will just say "Red", instead of "Apply filter: Red", or "Remove filter: Red". This also means that assistive technologies cannot tell whether the filter is already applied or not.
2. Disabled options are not identified as such to assistive technologies.
3. Options are styled to use a checkbox to indicate whether the filter is applied or not. This resembles the checkbox form control closely enough that the behaviour of following a link could be unexpected behaviour for clicking the checkbox.

**Describe the solution you'd like**

1. Add an `aria-label` to facet options indicating what will happen when following the link.
2. Add a `aria-disabled="true"` and `aria-role="link"` attributes to disabled options.
3. Explore alternative designs for filter links that communicate the state of the filter while still resembling a link.

**Additional context**
These issues are only relevant to the Facet widget, and not Instant Results, which uses form controls for filters.

**Comment by felipeelia** [comment]
Closed by #2607
